test/unit/custom_index_test.lua

```diff
@@ -197,11 +197,14 @@ function g.test_tree_index_multikey(cg)
     task:kill()
 end
 
-function g.test_memtx_tree_functional_index_broken(cg)
-    t.skip_if(_TARANTOOL < "2" or not helpers.memtx_func_index_is_broken(),
-              "Unsupported Tarantool version")
-    t.skip_if(cg.params.index_type ~= 'TREE', 'Unsupported index type')
+function g.test_memtx_tree_functional_index_broken_error(cg)
     t.skip_if(cg.params.engine == 'vinyl', 'Unsupported engine')
+    t.skip_if(cg.params.index_type ~= 'TREE', 'Unsupported index type')
+    t.skip_if(cg.space.index["functional_index"] == nil,
+        "Functional indexes are not supported by Tarantool")
+    t.skip_if(helpers.memtx_func_index_is_supported(),
+        "Doesn't blocked by https://github.com/tarantool/tarantool/issues/6786")
```

Comment:
Nit: I would say something like 'no error expected with a functional index on this tarantool version, correctness is verified by another test case'. Up to you.